lib/eigen_matrix_base_plugin.hpp

```diff
@@ -0,0 +1,4 @@
+template <typename OtherDerived>
```

Comment:
This piece of code is Eigen-specific, and I feel it's a bit awkward to put it here. How about you add some comment (e.g., mentioning https://eigen.tuxfamily.org/dox/TopicCustomizing_Plugins.html) to explain, or put it in a dedicated folder?

Reply:
I have just added a comment above it.

I feel it being a bit strange if I use a dedicated folder holding only one file.